typescript/mainLogic/services/service_context.ts

```diff
@@ -0,0 +1,157 @@
+interface GeneratedFileFilepaths { //TODO Chose to use interface to represent struct, will need to test to see if this hypothesis works
+	AbsoluteFilepathOnTestsuiteContainer: string;
+	AbsoluteFilepathOnServiceContainer:   string;
+}
+
+// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+interface ServiceContext {
+	client:                                   core_api_bindings_js_grpc.ApiContainerServiceClient;
+	serviceId:                                ServiceID;
+	ipAddress:                                string;
+	testVolumeMountpointOnTestsuiteContainer: string;
+	testVolumeMountpointOnServiceContainer:   string;
+}
+
+function NewServiceContext (
+		client: core_api_bindings_js_grpc.ApiContainerServiceClient,
+		serviceId: ServiceID,
+		ipAddress: string,
+		testVolumeMountpointOnTestsuiteContainer: string,
+		testVolumeMountpointOnServiceContainer: string) { //TODO - Return type for function, typescript doesn't need
+	let NewServiceContext = {
+        client:                                   client,
+        serviceId:                                serviceId,
+        ipAddress:                                ipAddress,
+        testVolumeMountpointOnTestsuiteContainer: testVolumeMountpointOnTestsuiteContainer,
+        testVolumeMountpointOnServiceContainer: testVolumeMountpointOnServiceContainer,
+    }
+    return NewServiceContext; //TODO return pointer in golang, but typescript's reference variables implciltly point into the heap
+}
+
+class ServiceContext { //TODO Should only be adding methods into a class, you can't link to a struct in typescript like in golang
+    
+    client: core_api_bindings_js_grpc.ApiContainerServiceClient;
+    serviceId: ServiceID;
+	ipAddress: string;
+	testVolumeMountpointOnTestsuiteContainer: string;
+	testVolumeMountpointOnServiceContainer: string;
+
+    constructor(
+        client: core_api_bindings_js_grpc.ApiContainerServiceClient,
+		serviceId: ServiceID,
+		ipAddress: string,
+		testVolumeMountpointOnTestsuiteContainer: string,
+		testVolumeMountpointOnServiceContainer: string) {
+            this.client = client;
+            this.serviceId = serviceId;
+            this.ipAddress = ipAddress;
+            this.testVolumeMountpointOnTestsuiteContainer = testVolumeMountpointOnTestsuiteContainer;
+            this.testVolumeMountpointOnServiceContainer = testVolumeMountpointOnServiceContainer;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public GetServiceID() {
+        return this.serviceId;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public GetIPAddress() {
+        return this.ipAddress;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public ExecCommand(command: string[]) {
+        let serviceId = this.serviceId;
+        let args = new core_api_bindings_js.ExecCommandArgs();
```

Comment:
is there a constructor `ExecCommandArgs` that takes in all the fields, by chance? Generally when coding you want your objects to be immutable, meaning that you set everything at constructor time and don't change it.

This has several benefits:
1) it's very clear what the state of the object is at any time - it's the state you constructed with 
2) immutable objects are always thread-safe (nothing is getting changed, so you don't have to worry about race conditions)
3) if you add a field to the object, the constructor will break with a compile error (rather than here, where you'd have to remember to add a new args.setXXXXXXXX() call)

Maybe the bindings generation library doesn't allow this (which would be sad), but if it does we should def use it